src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,38 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        goToX(robot, toX);
+        while (robot.getX() != toX) {
+            robot.stepForward();
+        }
+        goToY(robot, toY);
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
+    }
+
+    private void goToX(Robot robot, int toX) {
+        if (robot.getX() > toX) {
+            while (robot.getDirection() != Direction.LEFT) {
+                robot.turnLeft();
+            }
+        } else {
+            while (robot.getDirection() != Direction.RIGHT) {
+                robot.turnRight();
+            }
+        }
+    }
+
+    private void goToY(Robot robot, int toY) {
```

Comment:
You are not moving robot inside this method(just turning it in one place) so it's better to reconsider name for it